ACPs/75_consensus_proofs/75-consensus-proofs.md

```diff
@@ -0,0 +1,87 @@
+
+If ProposerVM blocks were produced with a proof of the P-Chain block to be verified against instead of just its height, verifiers could continue making progress by requesting any block they had not accepted yet from another peer and verifying their corresponding acceptance proofs. If a block’s proof is valid, the blocks can be executed in-order locally to finally verify the proposed subnet block.
+
+Figure 1: A Validator is verifying a subnet’s block `Z` which references an unknown P-Chain block `C` in its block header
+
+![figure 1](./1.jpg)
+
+Figure 2: A Validator requests the blocks and proofs for `B` and `C` from a peer
+
+![figure 2](./2.jpg)
+
+Figure 3: The Validator accepts the P-Chain blocks and is now able to verify `Z`
+
+![figure 3](./3.jpg)
+
+## Specification
+
+Note: The following is pseudocode and is provided only as visual aid.
+
+### ProposerVM
+
+The block header will require an update to include a Proof of the parent block
+to be verified at corresponding to the existing height field.
+
+```diff
+type BlockHeader struct {
+    PChainHeight uint64
+    Timestamp    time.Time
+    Proposer     ids.NodeID
++   Proof        []byte
+  }
+```
+
+### P2P
+
+The `Ancestors` message is replaced with the `containers`
+field which will supply a set of containers with their corresponding proofs.
+
+```diff
+message Ancestors {
+    bytes chain_id = 1;
+    uint32 request_id = 2;
+-   repeated bytes containers = 3;
++   repeated Container containers = 4;
+}
+```
+
+The `Container` message is used to send a container with its corresponding
+consensus proof.
+
+```diff
++ message Container {
++   bytes container = 1;
++   bytes proof = 2;
++ }
+```
+
+## Security Considerations
+
+Nodes that bootstrap using state sync may not have the entire history of the
+P-Chain and therefore will not be able to provide the entire history for a block
+that referenced in a block they propose. This would be needed to unblock a node that is attempting to fast-forward their P-Chain, as they require the entire ancestry between their current accepted tip and the block they are attempting to forward to. This is considered to be acceptable as
+nodes will be expected to maintain some minimum of some recent state. An alternative is to make a churn assumption and validate the proposed block's proof with a stale validator set.
```

Comment:
May also want to note that this is `ok` because it is assumed that `some` node has the data and with enough retries they will eventually get it? (unless we want to talk about how nodes should only keep recent blocks).